### ISSUE TITLE
Missing symlink causes error with linchpin fetch

### ISSUE THREAD

**Issue by herlo** [CentOS-PaaS-SIG/linchpin#340]
As unintended consequences go. I discovered this issue when testing out linchpin fetch. In one of my workspaces, I had been recently tinkering with credentials. In this case, the idea that a google cloud credential file could exist elsewhere and be linked. I might see this happen elsewhere with shared hooks or the like.

```$ ls -l /home/herlo/lp_workspace/credentials/gcloud/config.json
lrwxrwxrwx. 1 herlo herlo 23 May  4 15:42 /home/herlo/lp_workspace/credentials/gcloud/config.json -> /etc/gcloud/config.json
```
The above is the path that was linked. The /etc/gcloud/config.json file does not exist.

```
$ linchpin -w /tmp/workspace fetch layouts file:///home/herlo/lp_workspace/
Traceback (most recent call last):
  File "/home/herlo/Projects/virtualenvs/fetch/bin/linchpin", line 11, in <module>
    load_entry_point('linchpin', 'console_scripts', 'linchpin')()
  File "/home/herlo/Projects/virtualenvs/fetch/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/herlo/Projects/virtualenvs/fetch/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/herlo/Projects/virtualenvs/fetch/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/herlo/Projects/virtualenvs/fetch/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/herlo/Projects/virtualenvs/fetch/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/herlo/Projects/virtualenvs/fetch/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/home/herlo/Projects/virtualenvs/fetch/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/herlo/Projects/centos/git/linch-pin/linchpin/__init__.py", line 265, in fetch
    lpcli.lp_fetch(remote, ''.join(fetch_type), root)
  File "/home/herlo/Projects/centos/git/linch-pin/linchpin/api/__init__.py", line 459, in lp_fetch
    fetch_class.fetch_files()
  File "/home/herlo/Projects/centos/git/linch-pin/linchpin/api/fetch/fetch_local.py", line 46, in fetch_files
    self.copy_dir(path, td)
  File "/home/herlo/Projects/centos/git/linch-pin/linchpin/api/fetch/fetch.py", line 78, in copy_dir
    shutil.copy2(s, d)
  File "/usr/lib64/python2.7/shutil.py", line 130, in copy2
    copyfile(src, dst)
  File "/usr/lib64/python2.7/shutil.py", line 82, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: u'/home/herlo/lp_workspace/credentials/gcloud/config.json'
```

**Comment by agharibi** [comment]
@herlo I was able to replicate the error but it didn't give me an IOError 2. I fixed my issue but would you mind showing me the workspace that you were trying to fetch?

**Comment by herlo** [comment]
@agharibi The source workspace looks like this: 

```
$ tree /home/herlo/lp_workspace//home/herlo/lp_workspace/
├── credentials
│   ├── config.ini -> /etc/ec2/config.ini # (broken symlink)
│   ├── ec2
│   ├── gcloud
│   │   └── config.json -> /etc/gcloud/config.json # (broken symlink)
│   └── openstack
├── hooks
│   ├── ansible
│   │   └── ping
│   │       ├── dummy_ping.yaml
│   │       └── linchpin.log
│   └── shell
│       └── database
│           ├── init_db.sh
│           └── setup_db.sh
├── inventories
│   ├── dummy_cluster.inventory
│   ├── libvirt.inventory
│   └── os-test.inventory
├── layouts
│   ├── dummy-layout.yml
│   ├── example2-layout.yml
│   ├── example-layout.yml
│   └── openstack-test.yml
├── linchpin.log
├── PinFile
├── resources
│   ├── dummy_cluster.output
│   ├── libvirt.output
│   └── os-test.output
└── topologies
    ├── dummy-cluster.yml
    ├── example2-topology.yml
    ├── example-topology.yml
    └── openstack-cluster.yml

```
One of the noticeable things here is that I was trying to just fetch the layouts, but credentials is where the error occurred. I get why. Skipping broken links should be simple enough. Probably just a pass on the raised error would be good enough to fix this issue. We can't be expected to copy files that don't exist.

**Comment by agharibi** [comment]
Closed by #342
